modules/transports/core/vfs/src/main/java/org/apache/synapse/transport/vfs/VFSTransportListener.java

```diff
@@ -1194,5 +1208,149 @@ public int compare(FileObject o1, FileObject o2) {
             }
             return lDiff.intValue();
         }
-    }  
+    }
+
+    /**
+     * Verifies if the given md5 is the md5 of an Empty File
+     * @param entry current poll
+     * @param md5 of the given file
+     * @return true if configuration is set and file is empty
+     */
+    private boolean isFileEmpty(PollTableEntry entry, String md5) {
+        if (entry.isCheckSizeIgnoreEmpty()) {
+            return EMPTY_MD5.equals(md5);
+        }
+        return false;
+    }
+    
+    /**
+     * This Function calculates the MD5 Hash of the FileObject, waits
+     * checkSizeInterval [ms] time, and calculates the MD5 Hash again. If they
+     * are the same, the file is finished uploading and can be consumed.
+     *
+     * @param entry current poll
+     * @param child the fileobject currently read
+     * @return if file is empty or the filesize is changing
+     */
+    private boolean isFileStillChangingSize(PollTableEntry entry, FileObject child, String md5) {
+        //check if the lock mechanism is activated
+        if (!entry.hasCheckSizeInterval()) {
+            //not checking the file size changing
+            return false;
+        }
+        try {
+            //get interval time
+            String checkSizeIntervalString = entry.getCheckSizeInterval();
+            Long checkSizeInterval = Long.valueOf(checkSizeIntervalString);
+
+            //wait interval time 
+            log.debug("Check if file is still uploading. Now sleep "+checkSizeInterval+" ms");
+            Thread.sleep(checkSizeInterval);
+            //get second MD5
+            //clear cache and refresh
+            fsManager.getFilesCache().close();
+            child.refresh();
+            String md5AfterSleep = getMD5Checksum(child);
+            //System.out.println("After " + checkSizeInterval + "ms second size: " + md5B);
```

Comment:
Shall we remove this line if not needed or move to a debug log?

Reply:
Yes sure, the line https://github.com/wso2/wso2-synapse/pull/2095/files/dea5cb6ba6c5b8bfcf8e8b06ad7b858f2e80b19d#diff-8258f01ab96fc50493ea56c161dd25a7ca9f8835cfd76ccfab590716cb32ec1dR1254 should not be present